msprime/provenance.py

```diff
@@ -139,6 +139,11 @@ def default(self, obj):
                 'dtype': obj.dtype.str,
             }
 
+        elif isinstance(obj, bytes):
```

Comment:
I think we can drop these changes to provenance, they aren't needed AFAIK.

Reply:
Gee, I put them in in response to errors. I'll try taking them out again?

Reply:
Yes, I think so. I was thrashing around doing similar things before I figured out the right way to do ``asdict()`` (or at least, I realised that Ben had been doing it the right way before I broke it!)